workflow/artifacts/s3/s3.go

```diff
@@ -57,30 +71,42 @@ func (s3Driver *ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string)
 			s3cli, err := s3Driver.newS3Client(ctx)
 			if err != nil {
 				log.Warnf("Failed to create new S3 client: %v", err)
-				return false, nil
+				if isTransientS3Err(err) {
+					return false, nil
+				}
+				return false, fmt.Errorf("failed to create new S3 client: %v", err)
 			}
 			origErr := s3cli.GetFile(inputArtifact.S3.Bucket, inputArtifact.S3.Key, path)
 			if origErr == nil {
 				return true, nil
 			}
 			if !argos3.IsS3ErrCode(origErr, "NoSuchKey") {
-				log.Warnf("Failed get file: %v", origErr)
-				return false, nil
+				log.Warnf("Failed to get file: %v", origErr)
```

Comment:
You can remove this redundant warning that's the same as the error returned below. Same in other places too.

Reply:
My original thought is that: if the error is transient, `wait.ExponentialBackoff` will return a `wait.ErrWaitTimeout`. In this case, UI/CLI will only show a timeout error. Need logs to further check what the transient error actually is.

For now I moved the logging to the transient error block: 51c917569166aabfefc3f6983260eb2736d0e896. Will wait for our discussion in https://github.com/argoproj/argo-workflows/pull/5579#discussion_r606054053 to make further changes. Thanks!